samples/v2/component_with_optional_inputs.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional
+
+from kfp import compiler
+from kfp import dsl
+from kfp.dsl import component
+
+
+@component
+def component_op(
+    input1: str = 'default value',
+    input2: Optional[str] = None,
+    input3: Optional[str] = None,
```

Comment:
Would `None` have the same behavior as `''` (this may answer Chen's questions below)? If not, should we add an input with default value `''` to verify the behavior is as expected?

Reply:
Thank you! I don't think `None` is the same as empty string `''` in Python, and the input type `string` is covered in `input1`.